src/mongoc/mongoc-client-pool.c

```diff
@@ -17,6 +17,7 @@
 
 #include "mongoc-counters-private.h"
 #include "mongoc-client-pool.h"
+#include "mongoc-client-pool-private.h"
```

Comment:
mongoc-client-pool-private.h includes mongoc-client-pool.h, so you can just include the private header here.